pkg/svi/grpc.go

```diff
@@ -1,278 +1,195 @@
-		err := status.Errorf(codes.NotFound, "unable to find key %s", obj.Spec.LogicalBridge)
-		return nil, err
-	}
-	// fetch object from the database
-	vrf := new(pb.Vrf)
-	ok, err = s.store.Get(obj.Spec.Vrf, vrf)
-	if err != nil {
-		fmt.Printf("Failed to interact with store: %v", err)
-		return nil, err
-	}
-	if !ok {
-		err := status.Errorf(codes.NotFound, "unable to find key %s", obj.Spec.Vrf)
-		return nil, err
-	}
-	// configure netlink
-	if err := s.netlinkDeleteSvi(ctx, in, bridgeObject, vrf); err != nil {
-		return nil, err
-	}
-	// delete from FRR
-	vrfName := path.Base(vrf.Name)
-	vid := uint16(bridgeObject.Spec.VlanId)
-	vlanName := fmt.Sprintf("vlan%d", vid)
-	if err := s.frrDeleteSviRequest(ctx, obj, vrfName, vlanName); err != nil {
-		return nil, err
+		if !in.AllowMissing {
+			err = status.Errorf(codes.NotFound, "unable to find key %s", in.Name)
+			log.Printf("DeleteSvi(): Svi with id %v: Not Found %v", in.Name, err)
+			return nil, err
+		}
+		return &emptypb.Empty{}, nil
 	}
-	// remove from the Database
-	delete(s.ListHelper, obj.Name)
-	err = s.store.Delete(obj.Name)
-	if err != nil {
+
+	if err := s.deleteSvi(in.Name); err != nil {
+		log.Printf("DeleteSvi(): Svi with id %v, Delete Svi from DB failure: %v", in.Name, err)
 		return nil, err
 	}
+
 	return &emptypb.Empty{}, nil
 }
 
-// UpdateSvi updates an VLAN
-func (s *Server) UpdateSvi(ctx context.Context, in *pb.UpdateSviRequest) (*pb.Svi, error) {
+// UpdateSvi updates a Svi
+func (s *Server) UpdateSvi(_ context.Context, in *pb.UpdateSviRequest) (*pb.Svi, error) {
 	// check input correctness
 	if err := s.validateUpdateSviRequest(in); err != nil {
+		log.Printf("UpdateSvi(): validation failure: %v", err)
 		return nil, err
 	}
 	// fetch object from the database
-	svi := new(pb.Svi)
-	ok, err := s.store.Get(in.Svi.Name, svi)
+	sviObj, err := s.getSvi(in.Svi.Name)
 	if err != nil {
-		fmt.Printf("Failed to interact with store: %v", err)
-		return nil, err
-	}
-	if !ok {
-		// TODO: introduce "in.AllowMissing" field. In case "true", create a new resource, don't return error
-		err := status.Errorf(codes.NotFound, "unable to find key %s", in.Svi.Name)
-		return nil, err
-	}
-	// use netlink to find VlanId from LogicalBridge object
-	bridgeObject := new(pb.LogicalBridge)
-	ok, err = s.store.Get(svi.Spec.LogicalBridge, bridgeObject)
-	if err != nil {
-		fmt.Printf("Failed to interact with store: %v", err)
-		return nil, err
-	}
-	if !ok {
-		err := status.Errorf(codes.NotFound, "unable to find key %s", svi.Spec.LogicalBridge)
-		return nil, err
+		if err != infradb.ErrKeyNotFound {
+			log.Printf("UpdateSvi(): Failed to interact with store: %v", err)
+			return nil, err
+		}
+		if !in.AllowMissing {
+			err = status.Errorf(codes.NotFound, "unable to find key %s", in.Svi.Name)
+			log.Printf("UpdateSvi(): Svi with id %v: Not Found %v", in.Svi.Name, err)
+			return nil, err
+		}
+
+		log.Printf("UpdateSvi(): Svi with id %v is not found so it will be created", in.Svi.Name)
```

Comment:
Should the Update really do an auto-create if the port is not found or should it just return not found? (see port/grpc.go comment)

Reply:
Answered on that above